projects/scion/toolkit/uuid/src/uuid.util.ts

```diff
@@ -9,26 +9,36 @@
  */
 
 /**
- * Allows generating pseudo-random identifiers.
+ * Fast UUID generator, RFC4122 version 4 compliant.
+ *
+ * Usage: UUID.randomUUID()
  */
-export class UUID {
+function* generateUUID(): IterableIterator<string> {
+  const lut = [];
+  for (let i = 0; i < 256; i++) {
+    lut[i] = (i < 16 ? '0' : '') + (i).toString(16);
+  }
+
+  let uuid;
+  while (true) {
+    const d0 = Math.random() * 0xffffffff | 0;
+    const d1 = Math.random() * 0xffffffff | 0;
+    const d2 = Math.random() * 0xffffffff | 0;
+    const d3 = Math.random() * 0xffffffff | 0;
 
-  private constructor() {
+    uuid = lut[d0 & 0xff] + lut[d0 >> 8 & 0xff] + lut[d0 >> 16 & 0xff] + lut[d0 >> 24 & 0xff] + '-' +
+      lut[d1 & 0xff] + lut[d1 >> 8 & 0xff] + '-' + lut[d1 >> 16 & 0x0f | 0x40] + lut[d1 >> 24 & 0xff] + '-' +
+      lut[d2 & 0x3f | 0x80] + lut[d2 >> 8 & 0xff] + '-' + lut[d2 >> 16 & 0xff] + lut[d2 >> 24 & 0xff] +
+      lut[d3 & 0xff] + lut[d3 >> 8 & 0xff] + lut[d3 >> 16 & 0xff] + lut[d3 >> 24 & 0xff];
+
+    yield uuid;
   }
+}
 
-  /**
-   * Generates a 'pseudo-random' identifier.
-   */
-  public static randomUUID(): string {
-    let now = Date.now();
-    if (typeof window !== 'undefined' && typeof window.performance !== 'undefined' && typeof window.performance.now === 'function') {
-      now += performance.now(); // use high-precision timer if available
-    }
+const uuidIterator = generateUUID();
 
-    return 'xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx'.replace(/[xy]/g, char => {
-      const random = (now + Math.random() * 16) % 16 | 0;
-      now = Math.floor(now / 16);
-      return (char === 'x' ? random : (random & 0x3 | 0x8)).toString(16);
-    });
+export namespace UUID {
+  export function randomUUID(): string {
```

Comment:
Please add a JSDoc to the namespace and method. We have the convention for all public API to have a description.

E.g.: `Generates a UUID compliant with the RFC 4122 version 4 standard.`

Reply:
Added JSDoc to method. I added re-export to `public_api.ts` as `export * as UUID from './uuid.util';` and eliminated `namespace`. Namespaces are outdated constructs and not recommended from TypeScript itself. They generate IIFE (immediately invoked function expression) which are executed at load time and not tree-shakeable as well.

Everything works fine. You can use the functionality in the same manner as `UUID.randomUUID()`.
If you still want to have namespaces, I can revert re-exporting and add `namespace` again.